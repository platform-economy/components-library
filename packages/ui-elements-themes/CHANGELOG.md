# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
