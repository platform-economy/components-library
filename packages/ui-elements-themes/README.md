# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
