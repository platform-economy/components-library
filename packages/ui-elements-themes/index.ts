export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
