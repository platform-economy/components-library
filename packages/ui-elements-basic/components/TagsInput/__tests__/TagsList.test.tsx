import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

import { withTestTheme } from '@relayr/ui-elements-themes';

// Component
import { TagsList } from '../index';

enzyme.configure({ adapter: new Adapter() });

const sourceTags = [{ id:'1', title: 'test' }];

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('TagsInput', () => {
      it('should render tagsList with empty props', () => {
        const tree = TestRenderer
            .create(withTestTheme(
                <TagsList options={[]} />))
            .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render tagsList with props', () => {
        const tree = TestRenderer
            .create(withTestTheme(
                <TagsList options={sourceTags} />))
            .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render tagsList with props and loading', () => {
        const tree = TestRenderer
            .create(withTestTheme(
                <TagsList options={sourceTags} loading={true}/>))
            .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render tagsList with empty props and loading', () => {
        const tree = TestRenderer
            .create(withTestTheme(
                <TagsList options={[]} loading={true}/>))
            .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
