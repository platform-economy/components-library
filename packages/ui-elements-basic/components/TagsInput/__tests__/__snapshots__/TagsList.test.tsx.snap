// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components TagsInput should render tagsList with empty props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  min-height: 35px;
  width: 100%;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.23);
  background-color: #surface;
  color: #text;
  cursor: text;
  border: none;
  margin: 0;
  padding: .2em;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 li {
  margin-left: .5em;
  margin-top: 2px;
  list-style-type: none;
}

.c0 .c1 {
  margin-left: auto;
  padding: 5px;
  min-height: 25px;
}

.c0 span.noTagsMsg {
  color: #text;
  margin-left: 1em;
  font-size: .75em;
}

.c0 .loadingSpinner {
  margin: 2px;
}

<ul
  className="c0"
>
  <span
    className="noTagsMsg"
  />
</ul>
`;

exports[`ui-elements-basic Components TagsInput should render tagsList with empty props and loading 1`] = `
.c1 {
  height: 1em;
  width: 1em;
  border-radius: 50%;
  border: .2em solid #text_inverted;
  border-top: .2em solid #active;
  -webkit-animation: hmHMaM 1s infinite linear;
  animation: hmHMaM 1s infinite linear;
  box-sizing: border-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  min-height: 35px;
  width: 100%;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.23);
  background-color: #surface;
  color: #text;
  cursor: text;
  border: none;
  margin: 0;
  padding: .2em;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 li {
  margin-left: .5em;
  margin-top: 2px;
  list-style-type: none;
}

.c0 .c2 {
  margin-left: auto;
  padding: 5px;
  min-height: 25px;
}

.c0 span.noTagsMsg {
  color: #text;
  margin-left: 1em;
  font-size: .75em;
}

.c0 .loadingSpinner {
  margin: 2px;
}

<ul
  className="c0"
>
  <span
    className="noTagsMsg"
  />
  <div
    className="loadingSpinner c1"
  />
</ul>
`;

exports[`ui-elements-basic Components TagsInput should render tagsList with props 1`] = `
.c2 {
  min-height: 1.25em;
  padding: 0.125em 1em;
  background: #foreground;
  color: #text;
  border: none;
  line-height: 1.25em;
  border-radius: .75em;
  list-style-type: none;
  font-weight: normal;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
}

.c2 > span,
.c2 div,
.c2 i,
.c2 a {
  font-size: .75em;
}

.c2:hover {
  color: #text_highlight;
  background: #active_hovered;
}

.c2:active {
  color: #text_highlight;
  background: #active_lowered;
}

.c2:focus {
  color: #text_highlight;
  background: #active_hovered;
  border: none;
  outline: transparent;
}

.c5 {
  font-size: .75em;
  font-weight: bold;
  outline: none;
  text-align: center;
  vertical-align: middle;
  min-height: 35px;
  border: none;
  border-radius: 3px;
  padding: 5px 30px;
  -webkit-transition: 100ms all ease-in-out;
  transition: 100ms all ease-in-out;
  white-space: nowrap;
}

.c5::-moz-focus-inner {
  border: 0;
}

.c5 .icon-left {
  margin: 0 5px 0 0;
}

.c5 .icon-right {
  margin: 0 0 0 5px;
}

.c4 {
  color: #text_inverted;
  background-color: transparent;
}

.c4:hover,
.c4:focus {
  color: #text;
  cursor: pointer;
}

.c4:active {
  color: #active_lowered;
}

.c4:disabled {
  color: #inactive;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  min-height: 35px;
  width: 100%;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.23);
  background-color: #surface;
  color: #text;
  cursor: text;
  border: none;
  margin: 0;
  padding: .2em;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 li {
  margin-left: .5em;
  margin-top: 2px;
  list-style-type: none;
}

.c0 .c3 {
  margin-left: auto;
  padding: 5px;
  min-height: 25px;
}

.c0 span.noTagsMsg {
  color: #text;
  margin-left: 1em;
  font-size: .75em;
}

.c0 .loadingSpinner {
  margin: 2px;
}

.c1 {
  list-style-type: none;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <button
      className="Tag c2"
      onClick={[Function]}
      selected={false}
      title="test"
    >
      <span>
        test
      </span>
    </button>
  </li>
  <span
    className="noTagsMsg"
  />
  <button
    className="c3 c4 c5"
  >
    Select All
  </button>
</ul>
`;

exports[`ui-elements-basic Components TagsInput should render tagsList with props and loading 1`] = `
.c2 {
  min-height: 1.25em;
  padding: 0.125em 1em;
  background: #foreground;
  color: #text;
  border: none;
  line-height: 1.25em;
  border-radius: .75em;
  list-style-type: none;
  font-weight: normal;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
}

.c2 > span,
.c2 div,
.c2 i,
.c2 a {
  font-size: .75em;
}

.c2:hover {
  color: #text_highlight;
  background: #active_hovered;
}

.c2:active {
  color: #text_highlight;
  background: #active_lowered;
}

.c2:focus {
  color: #text_highlight;
  background: #active_hovered;
  border: none;
  outline: transparent;
}

.c3 {
  height: 1em;
  width: 1em;
  border-radius: 50%;
  border: .2em solid #text_inverted;
  border-top: .2em solid #active;
  -webkit-animation: hmHMaM 1s infinite linear;
  animation: hmHMaM 1s infinite linear;
  box-sizing: border-box;
}

.c6 {
  font-size: .75em;
  font-weight: bold;
  outline: none;
  text-align: center;
  vertical-align: middle;
  min-height: 35px;
  border: none;
  border-radius: 3px;
  padding: 5px 30px;
  -webkit-transition: 100ms all ease-in-out;
  transition: 100ms all ease-in-out;
  white-space: nowrap;
}

.c6::-moz-focus-inner {
  border: 0;
}

.c6 .icon-left {
  margin: 0 5px 0 0;
}

.c6 .icon-right {
  margin: 0 0 0 5px;
}

.c5 {
  color: #text_inverted;
  background-color: transparent;
}

.c5:hover,
.c5:focus {
  color: #text;
  cursor: pointer;
}

.c5:active {
  color: #active_lowered;
}

.c5:disabled {
  color: #inactive;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  min-height: 35px;
  width: 100%;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.23);
  background-color: #surface;
  color: #text;
  cursor: text;
  border: none;
  margin: 0;
  padding: .2em;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 li {
  margin-left: .5em;
  margin-top: 2px;
  list-style-type: none;
}

.c0 .c4 {
  margin-left: auto;
  padding: 5px;
  min-height: 25px;
}

.c0 span.noTagsMsg {
  color: #text;
  margin-left: 1em;
  font-size: .75em;
}

.c0 .loadingSpinner {
  margin: 2px;
}

.c1 {
  list-style-type: none;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <button
      className="Tag c2"
      onClick={[Function]}
      selected={false}
      title="test"
    >
      <span>
        test
      </span>
    </button>
  </li>
  <span
    className="noTagsMsg"
  />
  <div
    className="loadingSpinner c3"
  />
  <button
    className="c4 c5 c6"
  >
    Select All
  </button>
</ul>
`;
